### PR TITLE
Expose components needed for the ai plugin new interface

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
I'm exposing the ThreadView component and the AdvancedCreateComment component in the plugin interface. This is for the new interface for the AI plugin.

#### Ticket Link
```release-note
Exposing ThreadView and AdvancedCreateComment components in the webapp plugin exported components list.
```

### CODE REVIEW

**Comment by jespino** [comment]
It shouldn't be affecting QA in any way, I'm using it locally for my plugin and it is working. There is nothing that should break the application with this change, so we can skip QA for this.

**Comment by jespino** [comment]
Very good point, I can simplify a lot the component exposed, so, let me do it :)

**Comment by hmhealey** [comment]
CCing @larkox on this since he's been working on unifying and cleaning up the post textbox components

**Comment by larkox** [comment]
Short term, it should not affect much for my unification effort, but we should be really careful about the interface of the component from now on.

Extra comments in the components to make note of this and try to make people not change the interface would be nice, to avoid breaking plugins.

0/5

**Comment by jespino** [comment]
@larkox the current exported interface is just `placeholder` and `onSubmit`, so probably any change that we do under the hood, should be able to keep that behavior exposed.

**Comment by crspeller** [comment]
@jespino Not liking all the `|| ''` in the current implementation. Are we sure that all that is necessary or can some of the underlying implementation be fixed? Or can we pass appropriate values?

**Comment by jespino** [comment]
I going to make it way more isolated but skipping all the type checking.

File: webapp/channels/src/plugins/exported_create_post.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import type {PostDraft} from 'types/store/draft';
+
+type Props = {
+    placeholder?: string;
+    onSubmit: (draft: PostDraft) => void;
+}
+
+const ExportedCreatePost = ({placeholder, onSubmit}: Props) => {
+    const Component = AdvancedCreateComment as any
```
Comment (jespino) [inline webapp/channels/src/plugins/exported_create_post.tsx:16]: This is where the hacky part happens. Basically I'm accepting that I going to pass whatever I want to make the AdvancedCreateComment work.

File: webapp/channels/src/plugins/exported_create_post.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import type {PostDraft} from 'types/store/draft';
+
+type Props = {
+    placeholder?: string;
+    onSubmit: (draft: PostDraft) => void;
+}
+
+const ExportedCreatePost = ({placeholder, onSubmit}: Props) => {
+    const Component = AdvancedCreateComment as any;
```
Comment (crspeller) [inline webapp/channels/src/plugins/exported_create_post.tsx:16]: Not the best, but if the plan from @larkox is to clean up these components. We can fix these types then.

**Review by crspeller — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]

**Pull request merged**